### PR TITLE
Pipeline for GolangCI linter in compass-manager repository

### PR DESCRIPTION
**Pull request by mvshao**
**Description**

Changes proposed in this pull request:

- added pipeline for running golangci linter for compass-manager repository

### CODE REVIEW

File: prow/jobs/kyma-project/compass-manager/golang-linter.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      skip_report: false
+      decorate: true
+      path_alias: github.com/kyma-project/compass-manager
+      cluster: untrusted-workload
+      max_concurrency: 10
+      branches:
+        - ^main$
+      spec:
+        containers:
+          - image: "europe-docker.pkg.dev/kyma-project/dev/test-infra/golangci-lint:PR-7337"
```
Comment (Ressetkk) [inline prow/jobs/kyma-project/compass-manager/golang-linter.yaml:23]: Please don't use PR image here.

Comment (mvshao) [reply prow/jobs/kyma-project/compass-manager/golang-linter.yaml:23]: In the [Artifactory Registry](https://console.cloud.google.com/artifacts/docker/kyma-project/europe/dev/test-infra%2Fgolangci-lint?authuser=1), I can't find images that aren't PR's images. Could you please provide where I can find the right images?

File: prow/jobs/kyma-project/compass-manager/golang-linter.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+              - "make lint"
+            resources:
+              requests:
+                memory: 1Gi
+                cpu: 500m
+              limits:
+                memory: 3Gi
+                cpu: 1
+            securityContext:
+              privileged: true
```
Comment (Ressetkk) [inline prow/jobs/kyma-project/compass-manager/golang-linter.yaml:37]: This job doesn't need privileged mode. Remove it.

File: prow/jobs/kyma-project/compass-manager/golang-linter.yaml [hunk 3]
```diff
@@ -0,0 +1,37 @@
+presubmits: # runs on PRs
+  kyma-project/compass-manager:
+    - name: pre-main-compass-manager-lint
```
Comment (Ressetkk) [inline prow/jobs/kyma-project/compass-manager/golang-linter.yaml:3]: Job name should follow pattern `pull-repo-name`

**Review by Ressetkk — APPROVED** [review]

**Pull request merged**